mne/decoding/tests/test_time_gen.py

```diff
@@ -40,3 +44,163 @@ def test_time_generalization():
+    # Decim training steps
+    gat = GeneralizationAcrossTime(train_times={'step': .100})
+    gat.fit(epochs)
+    gat.score(epochs)
+    assert_equal(len(gat.scores_), 8)
+
+    # Test start stop training
+    gat = GeneralizationAcrossTime(train_times={'start': 0.090,
+                                                'stop': 0.250})
+    # predict without fit
+    assert_raises(RuntimeError, gat.predict, epochs)
+    gat.fit(epochs)
+    gat.score(epochs)
+    assert_equal(len(gat.scores_), 4)
+    assert_equal(gat.train_times['times_'][0], epochs.times[6])
+    assert_equal(gat.train_times['times_'][-1], epochs.times[9])
+
+    # Test diagonal decoding
+    gat = GeneralizationAcrossTime()
+    gat.fit(epochs)
+    scores = gat.score(epochs, test_times='diagonal')
+    assert_true(scores is gat.scores_)
+    assert_equal(np.shape(gat.scores_), (15, 1))
+
+    # Test generalization across conditions
+    gat = GeneralizationAcrossTime(predict_mode='mean-prediction')
+    gat.fit(epochs[0:6])
+    gat.predict(epochs[7:])
+    assert_raises(ValueError, gat.predict, epochs, test_times='hahahaha')
+    gat.score(epochs[7:])
+
+    svc = SVC(C=1, kernel='linear', probability=True)
+    gat = GeneralizationAcrossTime(clf=svc, predict_type='predict_proba',
+                                   predict_mode='mean-prediction')
+    gat.fit(epochs)
+
+    # sklearn needs it: c.f.
+    # https://github.com/scikit-learn/scikit-learn/issues/2723
+    # and http://bit.ly/1u7t8UT
+    assert_raises(ValueError, gat.score, epochs2)
+    gat.score(epochs)
+    scores = sum(scores, [])  # flatten
+    assert_true(0.0 <= np.min(scores) <= 1.0)
+    assert_true(0.0 <= np.max(scores) <= 1.0)
+
+    # test various predict_type
+    gat = GeneralizationAcrossTime(clf=svc, predict_type="predict_proba")
+    gat.fit(epochs)
+    gat.predict(epochs)
+    # check that 2 class probabilistic estimates are [p, 1-p]
+    assert_true(gat.y_pred_.shape[3] == 2)
+    gat.score(epochs)
+    # check that continuous prediction leads to AUC rather than accuracy
+    assert_true("roc_auc_score" in '%s' % gat.scorer_)
+
+    gat = GeneralizationAcrossTime(predict_type="decision_function")
+    # XXX Sklearn doesn't like non-binary inputs. We could binarize the data,
+    # or change Sklearn default behavior
+    epochs.events[:, 2][epochs.events[:, 2] == 3] = 0
+    gat.fit(epochs)
+    gat.predict(epochs)
+    # check that 2 class non-probabilistic continuous estimates are [distance]
+    assert_true(gat.y_pred_.shape[3] == 1)
+    gat.score(epochs)
+    # check that continuous prediction leads to AUC rather than accuracy
+    assert_true("roc_auc_score" in '%s' % gat.scorer_)
+
+    # Test that gets error if train on one dataset, test on another, and don't
+    # specify appropriate cv:
+    gat = GeneralizationAcrossTime()
+    gat.fit(epochs[0:6])
+    gat.predict(epochs[0:6])
+    assert_raises(ValueError, gat.predict, epochs)
```

Comment:
I see limited coverage. Only 79%:

mne.decoding.time_gen                338     56    165     48    79%   58, 190, 261, 338, 496, 530, 603, 757, 827-830, 866-889, 898-900, 929-992

a full function in 929-992 is not covered